File: socialuni-server/socialuni-community-sdk/src/main/java/com/socialuni/social/community/sdk/model/SocialuniTalkTabRO.java
package com.socialuni.social.community.sdk.model;

import lombok.Data;

/**
 * @author qinkaiyuan
 * @date 2020-05-23 17:21
 */
@Data
public class SocialuniTalkTabRO {
    private String name;
    private SocialuniTalkTabCircleRO circle;
    private String type;


    public SocialuniTalkTabRO(String name) {
        this.name = name;
    }
}

File: socialuni-js/packages/socialuni-api/socialuni-api-base/src/model/talk/SocialuniTalkTabRO.ts
import TalkVO from './TalkVO'
import SocialuniTalkTabCircleRO from "../community/circle/SocialuniTalkTabCircleRO";
import LoadMoreType from "socialuni-constant/constant/LoadMoreType";

export default class SocialuniTalkTabRO {
  name: string = null
  circle: SocialuniTalkTabCircleRO = null
  //有用的，需要根据此判断进入添加talk也面是否自定带入
  type: string = null
  talks: TalkVO[] = []
  //用户切换，或者tab切换时，为首次才加载
  firstLoad: boolean = true
  //用来区分是否为app内置tab，读取缓存过滤使用
  appDefaultTab: boolean = false
  scrollTop: number = -1
  pageScrollTop: number = -1

  constructor(name: string = null, type: string = null) {
    this.name = name
    this.type = type
  }
}
